lib/pfconfig/util.pm

```diff
@@ -34,55 +35,18 @@ our @EXPORT_OK = qw(
 Readonly our $undef_element => pfconfig::undef_element->new;
 
 sub fetch_socket {
+    use bytes;
     my ($socket, $payload) = @_;
-    
+
     # The line below will log any request to pfconfig for debugging purposes
     # As the logging cost even in trace is high, it's commented out
     #pfconfig::log::get_logger->info("Doing request to pfconfig with payload : '$payload'");
 
     # we ask the cachemaster for our namespaced key
-    print $socket "$payload\n";
-
-    # this will give us the line length to read
-
-    chomp( my $count = <$socket> ); 
-
-    # under some conditions we are receiving multiple lines.
-    # this under here should now fix it
-    use bytes;
-    my $line;
-    my $line_read = 0;
-    my $response  = '';
-    if($count =~ /\n/){
-        my @data = split(/\n/, $count);
-        my $length = scalar @data;
-        pfconfig::log::get_logger->warn("pfconfig has sent multiple lines with the count ($length)");
-        my $i = 0;
-        while($i < $length){
-            if($i == 0){
-                 $count = $data[0];
-            }
-            else{
-                # we chomp whatever we have to re-add it after so we hit all cases
-                $line = $data[$i];
-                $response .= $line . "\n";
-                $line_read += 1;
-            }
-            $i++;
-        }
-    }
-
-    # This is evil but we're getting lines with no content in them.
-    # This throws a warning that $line is undefined. 
-    # We workaround this by deactivating warnings when we read though the socket
-    no warnings;
-    while ( $line_read < $count ) {
-        chomp( $line = <$socket> );
-        $response .= $line . "\n";
-        $line_read += 1;
-    }
-    use warnings;
-    return $response;
+    $payload .= "\n";
```

Comment:
remove this, payload doesn't need to include this, it was only because it was reading lines